clang/lib/CodeGen/CGClass.cpp

```diff
@@ -2842,24 +2882,12 @@ void CodeGenFunction::EmitVTablePtrCheck(const CXXRecordDecl *RD,
       !CGM.HasHiddenLTOVisibility(RD))
     return;
 
-  SanitizerKind::SanitizerOrdinal M;
-  llvm::SanitizerStatKind SSK;
+  auto [M, SSK] = ParseCFITypeCheckKind(TCK);
   switch (TCK) {
```

Comment:
this switch is now a noop?

Reply:
Not quite. CodeGenFunction::ParseCFITypeCheckKind allows parsing CFITCK_ICall, but this switch statement will abort.

Reply:
then make that explicit rather than having this very long switch:

```
if (TCK == CFITCK_ICall) blah
```

but also unreachable is not abort, but triggers UB.

Reply:
We already crashed in ParseCFITypeCheckKind?

Reply:
> this switch is now a noop?

I've now simplified ParseCFITypeCheckKind in https://github.com/llvm/llvm-project/pull/139809/commits/e9c40b9ff66ea7476f2d750883df9c7c63b9fec6 such that the switch really is a no-op (and removed)